src/python/pants/backend/codegen/protobuf/python/rules.py

```diff
@@ -83,26 +93,55 @@ async def generate_python_from_protobuf(
         target_stripped_sources_request,
     )
 
-    input_digest = await Get(
-        Digest,
-        MergeDigests(
-            (
-                all_sources_stripped.snapshot.digest,
-                downloaded_protoc_binary.digest,
-                empty_output_dir,
-            )
-        ),
-    )
+    # To run the MyPy Protobuf plugin, we first install it with Pex, then extract the wheels and
+    # point Protoc to the extracted wheels with its `--plugin` argument.
+    extracted_mypy_wheels = None
+    if python_protobuf_subsystem.mypy_plugin:
+        mypy_pex = await Get(
+            Pex,
+            PexRequest(
+                output_filename="mypy_protobuf.pex",
+                internal_only=True,
+                requirements=PexRequirements(["mypy-protobuf==1.23"]),
+                interpreter_constraints=PexInterpreterConstraints(["CPython>=3.5,<3.9"]),
```

Comment:
This is the interpreter we run the pex in, unrelated to the compatibility of the code being typechecked?